components/menu/MenuData.tsx

```diff
@@ -0,0 +1,169 @@
+                  }
+              ` :
+            css`
+                  border-radius: 4px;
+                  background-color: rgba(var(--sidebar-text-rgb));
+                  padding: 1rem 0;
+                  transform: scale(0);
+                  opacity: 0;
+                  box-shadow: 0px 20px 32px rgba(0, 0, 0, 0.12);
+                  transition: opacity 300ms ease-in-out 0ms,
+                      transform 300ms ease-in-out 0ms;
+                  max-height: ${document.documentElement.clientHeight}px;
+                  min-width: 200px;
+                  overflow-y: auto;
+                  position: absolute;
+                  &.open {
+                      transform: scale(1);
+                      opacity: 1;
+                  }
+              `;
+    },
+);
+
+const MenuHeader = styled.div`
+    display: none;
+
+    &.isMobile {
+        display: flex;
+        justify-content: center;
+        border-bottom: 1px solid #e0e0e0;
+        height: 52px;
+    }
+
+    &.hasArrow {
+        display: grid;
+        grid-template-columns: 1fr auto 1fr;
+    }
+`;
+
+const MenuTitle = styled.div`
+    display: flex;
+    align-items: center;
+    color: var(--center-channel-color);
+`;
+
+const MenuData = React.forwardRef(
+    (props: MenuDataProps, ref: ForwardedRef<null>): JSX.Element => {
+        const {
+            title,
+            open,
+            groups,
+            trigger,
+            placement,
+            active,
+            isSubmenu,
+            closeSubmenu,
+            closeSubmenuDown,
+            isMobile,
+        } = props;
+
+        return (
+            <>
+                <MenuPopover
+                    isVisible={open}
+                    triggerRef={trigger}
+                    placement={placement}
+                    isMobile={isMobile}
+                >
+                    <MenuItems
+                        ref={ref}
+                        isMobile={isMobile}
+                        isSubmenu={Boolean(isSubmenu)}
+                        className={classNames({
+                            open,
+                            active,
+                            closeSubmenuDown,
+                        })}
+                    >
+                        <MenuHeader
+                            className={classNames({
+                                hasArrow: isSubmenu && isMobile,
+                                isMobile,
+                            })}
+                        >
+                            {isSubmenu && isMobile && (
+                                <i
+                                    className='icon icon-arrow-back-ios'
+                                    style={{
+                                        margin: '0 auto 0 23px',
+                                        color: '#3D3C40',
+                                    }}
+                                    onClick={closeSubmenu}
+                                />
```

Comment:
could we use the `IconArrowIcon` component from `@mattermost/compass-icons/components` here please?

Reply:
I am not sure about using compass since we don't know the future of the framework? Perhaps @zefhemel can advice?

Reply:
i am not suggesting using compass-components here, but the compass-icons package

Reply:
oops my bad